beekeeper-core/src/main/java/com/expediagroup/beekeeper/core/monitoring/TimedHousekeepingPathAspect.java

```diff
@@ -0,0 +1,71 @@
+/**
+ * This class is loosely based on {@link io.micrometer.core.aop.TimedAspect}. It has been adapted in order to create
+ * a custom tag for the timer metric that is registered each time it is called.
+ */
+package com.expediagroup.beekeeper.core.monitoring;
+
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.aspectj.lang.annotation.Around;
+import org.aspectj.lang.annotation.Aspect;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tags;
+import io.micrometer.core.instrument.Timer;
+
+import com.expediagroup.beekeeper.core.model.HousekeepingPath;
+
+/**
+ * Aspect for intercepting methods annotated with {@link TimedHousekeepingPath}.
+ *
+ * The method can have any number of arguments but a {@link HousekeepingPath} must be the first. Will add the fully
+ * qualified table name to the timer metric as a tag.
+ */
+@Aspect
+@Component
+public class TimedHousekeepingPathAspect {
```

Comment:
I don't see the implementation of the previous `Timed` aspect, but this one seems to be too specific and could be made a bit more generic.

The way I see it, the only difference is that it requires the object to contain the `tableName`.

I see two options:

a. Move the method to an interface that returns the tableName and change this aspect to a `TimedTableAspect`; or
b. Make the class implement a MetricTag interface and delegate the responsibility of returning what bit will be used to make the full metricTags to the implementing class.

The way it is right now, I see a potential of having to create very similar (but different) aspects for other metrics.

Reply:
b. sounds delicious, will give it a go to decouple this annotation from the `HousekeepingPath`

Reply:
This is now done. I've created a `Taggable` interface with one method `getMetricTag()`, which returns a `MetricTag` with the key for a tag and the tag itself. More decoupled/reusable now. Let me know what you think, cheers!